playground/app/directives/my-validator.directive.ts

```diff
@@ -18,7 +18,7 @@ export class MyValidatorDirective implements OnInit,AfterContentInit{
   }
   ngAfterContentInit(){
     console.log( this.todoSvc );
-    this.ngModelCtrl.$validators.myValidator = (viewValue,modelValue)=>{
+    this.ngModelCtrl.$validators["myValidator"] = (viewValue,modelValue)=>{
```

Comment:
I'm allergic to magic strings :D

It would be better to turn off TS compiler:
```typescript
(this.ngModelCtrl.$validators as any).myValidator = (viewValue, modelValue)=>{ /*.... */ );
```

Or even better we can extend angular type definitions:

```typescript
// ng-metadata.legacy.d.ts
declare module angular {
  export interface IModelValidators {
    myValidator: ModelValidator
  }
}
```

and then we can do:
```typescript
this.ngModelCtrl.$validators.myValidator = (viewValue, modelValue)=>{ /*.... */ );
```

would you be so kind and update it to first or second option?

anyway THANK YOU SO MUCH!!! you are awesome! :)